### PR TITLE
[GH-145] Add link to ephemeral message after uninstall

### LINKED ISSUE #145: After executing "/jira uninstall", link to the appropriate Jira Settings page in the resulting ephemeral message
SCENARIO 1: 
Repro: Successfully uninstall Jira Cloud via "/jira uninstall cloud <url>"
Observed: Ephemeral message:
```
Jira instance successfully disconnected. Go to **Settings > Apps > Manage Apps** to remove the application in your Jira Cloud instance.
```

EXPECTED: Ephemeral message, containing a link to Manage Apps page
```
Jira instance successfully disconnected. Go to [**Settings > Apps > Manage Apps**](...appropriate page link here...) to remove the application in your Jira Cloud instance.
```

----

SCENARIO 2: 
Repro: Successfully uninstall Jira Server via "/jira uninstall server <url>"
Observed: Ephemeral message:
```
Jira instance successfully disconnected. Go to **Settings > Applications > Application Links** to remove the application in your Jira Server or Data Center instance.
```

EXPECTED: Ephemeral message, containing a link to Manage Apps page
```
Jira instance successfully disconnected. Go to [**Settings > Applications > Application Links**](...appropriate page link here...) to remove the application in your Jira Server or Data Center instance.
```

----

If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Plugin: Jira" community channel](https://community.mattermost.com/core/channels/jira-plugin).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/) and our [Plugins Guide](https://developers.mattermost.com/extend/plugins/).

### PR DESCRIPTION
**Pull request by promulo**
#### Summary
This PR links the appropriate Jira settings page after `uninstall cloud|server` is successful. I took the liberty to perform some refactoring so only one function is used to perform the uninstall.

I have also added a test with basic cases. My original intent was to have more tests but I am not sure about the best way to setup a test for the `uninstall` command.

Thanks for reviewing!

#### Screenshots

image

image

#### Ticket Link
Fixes #145.

### CODE REVIEW

File: server/command.go [hunk 1]
```diff
@@ -418,60 +418,17 @@ If you see an option to create a Jira issue, you're all set! If not, refer to ou
...
-			"instance_installed": false,
-			"instance_type":      "",
-		},
-		&model.WebsocketBroadcast{},
-	)
-
-	const uninstallInstructions = `Jira instance successfully disconnected. Go to **Settings > Apps > Manage Apps** to remove the application in your Jira Cloud instance.`
-
-	return p.responsef(header, uninstallInstructions)
+	return executeUninstall(p, c, header, "cloud", args...)
```
Comment (levb) [inline server/command.go]: 1/5 change https://github.com/mattermost/mattermost-plugin-jira/blob/21a11bfc22191a399eefaee0813badcea1c7bfdc/server/command.go#L62-L63 to `"uninstall":    executeUninstall,` and check the parameter?

File: server/command.go [hunk 2]
```diff
@@ -517,9 +480,14 @@ func executeUninstallServer(p *Plugin, c *plugin.Context, header *model.CommandA
 		&model.WebsocketBroadcast{},
 	)
 
-	const uninstallInstructions = `Jira instance successfully disconnected. Go to **Settings > Applications > Application Links** to remove the application in your Jira Server or Data Center instance.`
+	var uninstallInstructions string
+	if instanceType == "cloud" {
+		uninstallInstructions = `Jira instance successfully disconnected. Go to [**Settings > Apps > Manage Apps**](%s/plugins/servlet/upm) to remove the application in your Jira Cloud instance.`
```
Comment (levb) [inline server/command.go]: 0/5 consider adding `GetManageAppsURL()` to `ji` and just use the method here?

File: server/command.go (hunk 1 above)
Comment (promulo) [reply server/command.go]: Hi @levb , thanks for reviewing. That was in fact the first thing that came to my mind but I wasn't sure if just omitting the `/cloud|server` part there would automatically make them show up in the `args` slice. Is that the case?

File: server/command.go (hunk 2 above)
Comment (promulo) [reply server/command.go]: Yeah, that makes sense. :+1:

File: server/command.go (hunk 1 above)
Comment (levb) [reply server/command.go]: Yep, see https://github.com/mattermost/mattermost-plugin-jira/blob/master/server/command.go#L120

**Comment by levb** [comment]
@promulo please ping us when you are done with these changes.

**Comment by promulo** [comment]
Hi @levb @aaronrothschild I've updated the PR after your feedback. Thanks for reviewing!

File: server/command.go [hunk 3]
```diff
@@ -442,67 +441,25 @@ func executeUninstallCloud(p *Plugin, c *plugin.Context, header *model.CommandAr
...
-	if err != nil {
-		return p.responsef(header, "%v", err)
-	}
-	if !authorized {
-		return p.responsef(header, "`/jira uninstall` can only be run by a System Administrator.")
-	}
-	if len(args) != 1 {
+	instanceType := args[0]
+	var ok bool
+	if instanceType == "cloud" {
```
Comment (levb) [inline server/command.go]: nit: a `switch args[0] {` would look better here, perhaps?

File: server/command.go [hunk 4]
```diff
@@ -517,9 +474,8 @@ func executeUninstallServer(p *Plugin, c *plugin.Context, header *model.CommandA
 		&model.WebsocketBroadcast{},
 	)
 
-	const uninstallInstructions = `Jira instance successfully disconnected. Go to **Settings > Applications > Application Links** to remove the application in your Jira Server or Data Center instance.`
-
-	return p.responsef(header, uninstallInstructions)
+	uninstallInstructions := `Jira instance successfully disconnected. Navigate to [**your app management URL**](%s) in order to remove the application from your Jira instance.`
```
Comment (levb) [inline server/command.go]: nit: maybe we should change `disconnected` to `uninstalled` to be consistent. @aaronrothschild ?

File: server/command.go (hunk 3 above)
Comment (promulo) [reply server/command.go]: It does, indeed.

**Comment by levb** [comment]
@aaronrothschild ping on PM review

**Comment by jfrerich** [comment]
@aaronrothschild little ping for review

**Comment by hanzei** [comment]
Thanks for this great contribution @promulo :+1:

**Pull request merged**